Tests/Filter/DateRangeFilterTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Patrick Landolt <patrick.landolt@artack.ch>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Tests\Filter;
+
+use Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQuery;
+use Sonata\DoctrineORMAdminBundle\Filter\DateRangeFilter;
+
+class DateRangeFilterTest extends \PHPUnit_Framework_TestCase
+{
+    public function testFilterEmpty()
+    {
+        $filter = new DateRangeFilter();
+        $filter->initialize('field_name', array('field_options' => array('class' => 'FooBar')));
+
+        $builder = new ProxyQuery(new QueryBuilder());
+
+        $filter->filter($builder, 'alias', 'field', null);
+        $filter->filter($builder, 'alias', 'field', '');
+        $filter->filter($builder, 'alias', 'field', 'test');
+        $filter->filter($builder, 'alias', 'field', false);
+
+        $filter->filter($builder, 'alias', 'field', array());
+        $filter->filter($builder, 'alias', 'field', array(null, 'test'));
+        $filter->filter($builder, 'alias', 'field', array('type' => null, 'value' => array()));
+        $filter->filter($builder, 'alias', 'field', array('type' => null, 'value' => array('start' => null, 'end' => null)));
+        $filter->filter($builder, 'alias', 'field', array('type' => null, 'value' => array('start' => '', 'end' => '')));
+
+        $this->assertEquals(array(), $builder->query);
```

Comment:
I changed it to `assertSame` due to an empty array() is the initial value of `QueryBuilder`